dali/python/nvidia/dali/plugin/numba/experimental/__init__.py

```diff
@@ -141,11 +142,11 @@ def _get_carray_eval_lambda(self, dtype, ndim):
             eval_string += "shape[{}]".format(i)
             eval_string += ", " if i + 1 != ndim else "), "
         eval_string += "dtype=np.{})".format(_to_numpy[dtype])
-        return njit(eval(eval_string))
+        return njit(ast.literal_eval(eval_string))
```

Comment:
Does that work?

``` The string or node provided may only consist of the following Python literal structures: strings, bytes, numbers, tuples, lists, dicts, sets, booleans, None and Ellipsis.```

Reply:
I'm quite sure it won't work. We're generating _code_ here. The key part is that it's as trusted as the program itself.

Reply:
Fixed